src/ubuntu/ubuntu_i2c_master.c

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <linux/i2c-dev.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/ioctl.h>
+#include <sys/select.h>
+#include <unistd.h>
+#include "common/cs_dbg.h"
+#include "mgos_i2c.h"
+#include "ubuntu_ipc.h"
+
+struct mgos_i2c {
+  struct mgos_config_i2c cfg;
+  int freq;
+  int read_timeout_ms;
+  int fd;
+  int dev_no;
+};
+
+static const char *mgos_i2c_dev_filename(struct mgos_i2c *c) {
+  static char fn[256];
+
+  if (!c) {
```

Comment:
style point: we don't use boolean comparison for pointers, so - `if (c == NULL)` please

Reply:
Done. I will have to unlearn this :) and go over the other code I submitted which consistently uses `if (!c)`. Updated here.